src/liboslexec/llvm_passes.h

```diff
@@ -37,9 +38,8 @@ namespace {
 // how instruction lowering happens then this pass may not be necessary.
 // Also if future LLVM version takes on the work of this optimization pass,
 // then it may be removed.
-template<int WidthT>
-class PreventBitMasksFromBeingLiveinsToBasicBlocks final
-    : public llvm::FunctionPass {
+
+template<int WidthT> class PreventBitMasksFromBeingLiveinsToBasicBlocks {
```

Comment:
I'm not sure I understand what's going on here. Why does this no longer inherit from llvm::FunctionPass?

Reply:
Only for the legacy pass manager do we need to inherit from `llvm::FunctionPass`. This class contains the implementation that is shared by `NewPreventBitMasksFromBeingLiveinsToBasicBlocks` and `LegacyPreventBitMasksFromBeingLiveinsToBasicBlocks`.